en/guides/typesafehttpbasic.md

```diff
@@ -0,0 +1,172 @@
+
+HTTP Basic authentication transmits credentials in an "Authorization" header as base64 encoded user ID/password pairs. Many clients also let you send the username and the password in the URL as follows:
+
+```
+https://username:password@www.example.com/
+```
+
+However some web browsers disable this for security reasons.
+
+This guide steps you through using `TypeSafeHTTPBasic`, a `TypeSafeCredentials` plugin for safe and easy HTTP basic authentication in Codable routes.
+
+## Adding TypeSafeHTTPBasic to your project
+
+1) Add `Kitura-CredentialsHTTP` to your `Package.swift` dependencies:
+```swift
+.package(url: "https://github.com/IBM-Swift/Kitura-CredentialsHTTP.git", from: "2.1.0"),
+```
+
+2) Add `CredentialsHTTP` to your Application targets:
+```swift
+.target(name: "Application", dependencies: [ "CredentialsHTTP", ...
+```
+
+3) Regenerate your Xcode project:
+```
+swift package generate-xcodeproj
+```
+
+4) Open your Xcode project and go to Sources >> Application >> Application.swift
+
+5) Add `CredentialsHTTP` to your import statements:
+```swift
+import CredentialsHTTP
+```
+
+## Defining a TypeSafeHTTPBasic object
+
+We will declare a struct which conforms to `TypeSafeHTTPBasic`. This will be initialized when our route is successfully authenticated and we will be able to access the authenticated user's id within our Codable route.
+
+1) Define the TypeSafeHTTPBasic Struct:
+
+At the bottom of `Application.swift`, Define a public struct called `MyBasicAuth` and set it to conforms to `TypeSafeHTTPBasic`:
+```swift
+public struct MyBasicAuth: TypeSafeHTTPBasic {
+
+}
+```
+2) Add protocol stubs:
+
+Xcode should appear with the message:
+`Type 'MyBasicAuth' does not conform to protocol 'TypeSafeCredentials'`
+
+Click fix to autogenerate the stubs and produce the below struct:
+```swift
+public struct MyBasicAuth: TypeSafeHTTPBasic {
+    public static var verifyPassword: ((String, String, @escaping (MyBasicAuth?) -> Void) -> Void)
+
+    public var id: String
+}
+```
+
+3) Inside MyBasicAuth, add an authentication dictionary:
+
+
+```swift
+public static let authenticate = ["username" : "password"]
+```
+
+__Note:__ In a real project, never store passwords in plain text!
+
+4) Define verifyPassword:
+
+The closure, `verifyPassword`, takes a username and password and returns a `MyBasicAuth` instance on success. We want to check if the password matches the user's stored password. On successful match, we initialise `MyBasicAuth` with an `id` equal to username.
+
+```swift
+public static var verifyPassword: ((String, String, @escaping (MyBasicAuth?) -> Void) -> Void) =
+   { userId, password, callback in
+       if let storedPassword = authenticate[userId], storedPassword == password {
```

Comment:
Can't we just show lines 85-89 here as they are the new content? (and you're showing the full struct below in step 5?).